src/js/structs/__tests__/JobTree-test.js

```diff
@@ -0,0 +1,115 @@
+let Job = require('../Job');
+let JobTree = require('../JobTree');
+
+describe('JobTree', function () {
+
+  describe('#constructor', function () {
+
+    beforeEach(function () {
+      this.instance = new JobTree({
+        id: '/group',
+        items: [
+          new JobTree({id: '/group/foo', items: []}),
+          {id: '/group/bar', items: []},
+          {id: '/group/alpha'},
+          new Job({id: '/group/beta'}),
+          {id: '/group/wibble/wobble'}
+        ],
+      });
+    });
+
+    it('defaults id to root tree (groups) id', function () {
+      let tree = new JobTree({items: []});
+      expect(tree.getId()).toEqual('/');
+    });
+
+    it('sets correct tree (groups) id', function () {
```

Comment:
how about `it(creates the group ids correctly)` and since we're at it how about adding 
```
expect(this.instance.getId()).toEqual('/group');
expect(this.instance.getItems()[0].getId()).toEqual('/group/foo');
```
?